packages/compartment-mapper/src/import-hook.js

```diff
@@ -43,12 +45,19 @@ export const makeImportHookMaker = (
       // third-party dependency.
       // The `moduleMapHook` captures all third-party dependencies.
       if (moduleSpecifier !== '.' && !moduleSpecifier.startsWith('./')) {
-        packageSources[moduleSpecifier] = {
-          exit: moduleSpecifier,
-        };
-        // Return a place-holder.
-        // Archived compartments are not executed.
-        return freeze({ imports: [], execute() {} });
+        if (exitModules[moduleSpecifier]) {
```

Comment:
To truly treat the values as meaningless, please don't ignore non-truthy keys:
```suggestion
        if (moduleSpecifier in exitModules) {
```

Reply:
> To truly treat the values as meaningless, please don't ignore non-truthy keys:

I went with `has(exitModules, moduleSpecifier` where:

```js
const { apply } = Reflect;
const { hasOwnProperty } = Object.prototype;
/**
 * @param {Record<string, never>} haystack
 * @param {string} needle
 */
const has = (haystack, needle) => apply(hasOwnProperty, haystack, [needle]);
```

To remove collisions up the prototype chain.